src/core/experimental/memory/memcpy.d

```diff
@@ -0,0 +1,401 @@
+/**
+ * Pure D replacement of the C Standard Library memcpy().
+ * There is an idiomatic-D interface, memcpy(), which is split into 3 overloads.
+ * One taking static types, one dynamic arrays and one static arrays.
+ * Also, there is available a C-like interface, the Dmemcpy() (which is named Dmemcpy
+ * for disambiguation with the C memcpy() a _similar_ interface) that
+ * is the classic (void*, void*, size_t) interface.
+ * N.B.: Both the memcpy() here and Dmemcpy() return nothing, contrary to the C Standard
+ * Library version.
+ * Source: $(DRUNTIMESRC core/experimental/memory/memcpy.d)
+ */
+module core.experimental.memory.memcpy;
+
+import core.internal.traits : isArray;
+
+/**
+ * Handle Static Types
+ * N.B.: No need for more sophisticated code for static types. The compiler
+ * knows better how to handle them in every target case.
+ *
+ * Params:
+ *  dst = Reference to memory destination to copy bytes to.
+ *  src = Reference to memory source to copy bytes from.
+ */
+pragma(inline, true)
+void memcpy(T)(ref T dst, ref const T src)
+if (!isArray!T)
+{
+    dst = src;
+}
+
+/**
+ * Handle Dynamic Types
+ *
+ * Params:
+ *  dst = Reference to destination dynamic array to copy bytes to.
+ *  src = Reference to source dynamic array to copy bytes from.
+ */
+void memcpy(T)(ref T[] dst, ref const T[] src)
+{
+    assert(dst.length == src.length);
+    void* d = cast(void*) dst.ptr;
+    const(void)* s = cast(const(void)*) src.ptr;
+    size_t n = dst.length * typeof(dst[0]).sizeof;
+    // Assume that there is no overlap.
+    pragma(inline, true);
```

Comment:
What does this mean?

Reply:
What? It shows me 4 lines. Please point me to the line.

Reply:
The one I tagged... `pragma(inline, true);`
That's a weird statement, what does it mean? What does it apply to?

Reply:
As far as I know, `pragma(inline, true)` instructs the compiler to inline the function call that follows.

Reply:
as a stand-alone statement on its own hanging there? What does it apply to?

Can you show the doco where you read that? If that exists I'll be very surprised (I've been asking for it for many years).

Reply:
It has effect, but not the effect I said. My mistake: https://dlang.org/spec/pragma.html#inline
It inlines the function it is _enclosed_ by.

Reply:
Oh wow. I had no idea you could do that. That's very weird.

Reply:
It seems weird to me as well..